Первая рабочая версия калькулятора

SergeiTakt opened the pull request.

Commit:
first commit: добавить первую рабочую версию калькулятора

ArturNurtdinov commented on src/main/java/Main.java:
```diff
@@ -1,6 +1,33 @@
+            calculator.add(product);
+            System.out.println("Товар успешно добавлен в калькулятор");
+
+            System.out.println("Хотите ли вы добавить еще один товар?");
```
Тут бы дописать, что в случае, если пользователь не хочет вводить еще товар - то ему нужно написать "завершить"

SergeiTakt replied:
🟢

ArturNurtdinov commented on src/main/java/calculator/Calculator.java:
```diff
@@ -0,0 +1,57 @@
+        }
+
+        double lastDigit = roundedPrice % 10;
+        if (lastDigit == 1.0) {
```
Числа с плавающей точкой нельзя сравнивать через `==`, т.к. существует погрешность в определённых порядках, вот здесь можно почитать про эту проблему - https://ru.hexlet.io/qna/java/questions/kak-pravilno-sravnivat-double-v-java?ysclid=lsey8ohru1864420997

SergeiTakt replied:
🟢

ArturNurtdinov commented on src/main/java/exception/KopecksFormatException.java:
```diff
@@ -0,0 +1,4 @@
+package exception;
+
+public class KopecksFormatException extends RuntimeException {
```
А почему именно RuntimeException, а не просто Exception? Хотелось бы и во время компиляции понимать, что из определённой функции может быть выброшен этот эксепшен и корректно его обрабатывать

SergeiTakt replied:
🟢

Commit:
fix: исправить ошибки, модернизировать код

В классе Calculator в методе getDeclensionByNumber(double price), чтобы избежать ошибок при сравнении чисел типа double, привел значение цены к типу int. Чтобы избежать переполнения int, сначала проверил фактическое значение.

 Класс KopecksFormatException унаследовал от Exception, чтобы исключение стало проверяемым.

Commit:
fix: исправить ошибки, модернизировать код

В классе Calculator в методе getDeclensionByNumber(double price), чтобы избежать ошибок при сравнении чисел типа double, привел значение цены к типу int. Чтобы избежать переполнения int, сначала проверил фактическое значение.

 Класс KopecksFormatException унаследовал от Exception, чтобы исключение стало проверяемым.

SergeiBrin commented:
Привет, Артур. Спасибо за ревью. Даже и не знал, что double нельзя сравнивать через == :)
Работу над ошибками провёл. Также внёс пару несущественных улучшений в код 😃

Commit:
fix: исправить ошибки, модернизировать код

В классе Calculator в методе getDeclensionByNumber(double price), чтобы избежать ошибок при сравнении чисел типа double, привел значения к типу int.

 Класс KopecksFormatException унаследовал от Exception, чтобы исключение стало проверяемым.

Commit:
Merge remote-tracking branch 'origin/dev' into dev

# Conflicts:
#	src/main/java/calculator/Calculator.java

Commit:
Merge remote-tracking branch 'origin/dev' into dev
